engine/src/main/java/org/hibernate/search/spatial/impl/DistanceCollector.java

```diff
@@ -147,8 +152,8 @@ public void setScorer(Scorer scorer) throws IOException {
 		@Override
 		public void collect(int doc) throws IOException {
 			final int absolute = docBase + doc;
-			double lat = coordinate( latitudeValues, doc );
-			double lon = coordinate( longitudeValues, doc );
+			Double lat = coordinate( docsWithLatitude, latitudeValues, doc );
```

Comment:
rather than changing the `coordinate` method, which is forcing you to return a `Double` rather than a `double`, you could check for this document to actually have both coordinates in advance, and if not skip the collection.